cmd/kubeadm/app/cmd/phases/selfhosting.go

```diff
@@ -48,5 +57,8 @@ func NewCmdSelfhosting() *cobra.Command {
 	}
 
 	cmd.Flags().StringVar(&kubeConfigFile, "kubeconfig", "/etc/kubernetes/admin.conf", "The KubeConfig file to use for talking to the cluster")
+	cmd.Flags().StringVar(&featureFlagsString, "feature-gates", featureFlagsString, "A set of key=value pairs that describe feature gates for alpha/experimental features. "+
```

Comment:
I tested this approach, but I'm not sure that the additional struct makes the code more readable (the final number of lines in init.go or selfhosting.go is nearly the same). 

However, if we decide to rework later, the implementation was:

in features.go

```go
// FeatureGatesFlag represent a variable backing a --feature-flag
type FeatureGatesFlag struct {
	featureList        *FeatureList
	featureGatesString string
}

// AddFeatureGatesToFlagSet adds --feature-flag for a specific FeatureList to a given FlagSet
func AddFeatureGatesToFlagSet(fs *pflag.FlagSet, featureList *FeatureList) *FeatureGatesFlag {
	f := &FeatureGatesFlag{featureList: featureList}
	fs.StringVar(&f.featureGatesString, "feature-gates", f.featureGatesString, "A set of key=value pairs that describe feature gates for various features. Options are:\n"+strings.Join(KnownFeatures(f.featureList), "\n"))
	return f
}

// GetFeatureGatesMap returns a Map containing selected features
func (f *FeatureGatesFlag) GetFeatureGatesMap() (map[string]bool, error) {
	return NewFeatureGate(f.featureList, f.featureGatesString)
}
```

in init.go or selfhosting.go

```go
var featureGatesFlag *features.FeatureGatesFlag

// in cobra Run() {
   var err error
   if cfg.FeatureFlags, err = featureGatesFlag.GetFeatureGatesMap(); err != nil {
      kubeadmutil.CheckErr(err)
   }
}

featureGatesFlag = features.AddFeatureGatesToFlagSet(cmd.Flags(), &features.InitFeatureGates)
```